.github/workflows/main.yml

```diff
@@ -7,9 +7,12 @@ on:
     branches: ["main"]
   pull_request:
     branches: ["main"]
+  workflow_dispatch:
```

Comment:
The `workflow_dispatch` event should be properly formatted as a mapping. It should be written as `workflow_dispatch: {}` to ensure it is recognized correctly by GitHub Actions.